### ISSUE TITLE
When using in `UIViewRepresentable`, update the `Binding` value does not trigger `updateUIView(:)` during unit test

### ISSUE THREAD

**Issue by dreampiggy** [nalexn/ViewInspector#6]
Demo `UIViewRepresentable` Code:
Just check https://github.com/SDWebImage/SDWebImageSwiftUI/blob/master/SDWebImageSwiftUI/Classes/AnimatedImage.swift :)

Demo Unit Test Code:

```swift
 func testAnimatedImageBinding() throws {
    let expectation = self.expectation(description: "AnimatedImage binding control")
    var binding = Binding<Bool>(wrappedValue: true)
    let imageView = AnimatedImage(name: "TestImage.gif", bundle: testImageBundle(), isAnimating: binding)
    let introspectView = imageView.introspectAnimatedImage { animatedImageView in
        if let animatedImage = animatedImageView.image as? SDAnimatedImage {
            XCTAssertEqual(animatedImage.animatedImageLoopCount, 0)
            XCTAssertEqual(animatedImage.animatedImageFrameCount, 5)
        } else {
            XCTFail("SDAnimatedImageView.image invalid")
        }
        XCTAssertTrue(animatedImageView.isAnimating)
        binding.wrappedValue = false
        binding.update()
        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
            XCTAssertFalse(binding.wrappedValue)
            XCTAssertFalse(imageView.isAnimating)
            XCTAssertFalse(animatedImageView.isAnimating) // <----- Failed on this line
            expectation.fulfill()
        }
    }
    _ = try introspectView.inspect(AnimatedImage.self)
    ViewHosting.host(view: introspectView)
    self.waitForExpectations(timeout: 5, handler: nil)
}
```

**Comment by dreampiggy** [comment]
This `introspectAnimatedImage` method is just a helper method using https://github.com/siteline/SwiftUI-Introspect to introspect the native UIView (UIImageView here). I want to test the case:

+ When binding value turn to false, the `updateUIView` from SwiftUI is called, then `UIImageView.stopAnimating()` is called, and then `UIImageView.isAnimating` turn to false.

**Comment by dreampiggy** [comment]
The toturial you post [here](https://github.com/nalexn/ViewInspector/blob/master/guide.md#views-using-binding), seems only showing the case:

+ When toggle the button state, we update the Binding to the value outside.

However, seems my case is:

+ When update the Binding value outside, should toggle the button state

**Comment by nalexn** [comment]
Hey @dreampiggy ,

Your use case should be possible to test, however, after you change the Binding value outside, you'd need to trigger the asynchronous inspection described [in the next section](https://github.com/nalexn/ViewInspector/blob/master/guide.md#views-using-state-environment-or-environmentobject).
You may use `inspect` with `delay` parameter defaulting to zero.

**Comment by dreampiggy** [comment]
This issue can be closed.

For `UIViewRepresentable`, you have to wrap that into a standalone native SwiftUI view. And it should have a closure that receive `Self` and send it outside. Because the structure is copied.

Using the new test code solve the problem:

https://gist.github.com/dreampiggy/0b83d31428b3bed0e5be7ffd7ac4aa02

**Comment by dreampiggy** [comment]
Founded a new issue related to SwiftUI itself ?

Why same code, SwiftUI's NSViewRepresentable call `updateNSView` once, but UIViewRepresentable call `updateUIView` twice 😅 

Not related to ViewInspector, but just a little curious.
